src/builder-manifest.c

```diff
@@ -2751,14 +2745,20 @@ builder_manifest_cleanup (BuilderManifest *self,
 
       if (self->appstream_compose && appdata_file != NULL)
         {
-          g_autofree char *basename_arg = g_strdup_printf ("--basename=%s", self->id);
-          g_print ("Running appstream-compose\n");
-          if (!appstream_compose (app_dir, error,
-                                  self->build_runtime ?  "--prefix=/usr" : "--prefix=/app",
-                                  "--origin=flatpak",
-                                  basename_arg,
-                                  self->id,
-                                  NULL))
+          g_autofree char *components_arg = g_strdup_printf ("--components=%s", self->id);
+          g_autofree char *app_root_path = g_file_get_path(app_root);
+          g_autofree char *result_root_arg = g_strdup_printf ("--result-root=%s", app_root_path);
+
+          g_print ("Running appstreamcli-compose\n");
+          if (!appstreamcli_compose (error,
+                                     "--prefix=",
+                                     "--origin=flatpak",
+                                     "--verbose",
+                                     "--print-report=short",
```

Comment:
You *may* want to increase verbosity for the issues found, but only if users are actually expected to read this. The `short` report will display the tag names of all the issues found, `on-error` (the default) will only display information if there was an error, and `full` will show the found issue tags and their explanatory texts for people who are less familiar with AppStream. This will increase verbosity quite a bit.
I think either `short` or `full` may be good options here.

Reply:
I don't think users are expected to do anything about compose output. We should probably add a separate validation step like Flathub does as part of its CI, but I guess it's out of scope here.